RELEASING.md

```diff
@@ -0,0 +1,60 @@
+# Releasing Federation
+
+This is a quick and opinionated set of commands for building and releasing a new version of the packages in this monorepo.
+
+#### Start a release PR
+
+1. `FEDERATION_RELEASE_VERSION=X.Y.Z` (e.g. `2.0.1` or `2.0.0-beta.3` or `2.0.0-preview.99`)
+1. `git fetch origin main:main` to fetch the latest `main` branch
+1. `git checkout -b "release-$FEDERATION_RELEASE_VERSION" main`
+1. Update `CHANGELOG.md` in each repo ([more info](#changelogs))
+1. `git add ':/*CHANGELOG.md'` to add only changelogs to the commit
+1. `git commit --message "Update CHANGELOGs for v$FEDERATION_RELEASE_VERSION release"`
+1. `git push -u origin "release-$FEDERATION_RELEASE_VERSION"`
+1. `open "https://github.com/apollographql/federation/compare/main...release-$FEDERATION_RELEASE_VERSION?quick_pull=1&title=Release+$FEDERATION_RELEASE_VERSION&template=APOLLO_RELEASE_TEMPLATE.md"` to open PR in Github
+
+#### Bump version and tag release
+
+1. `npx lerna version --no-push --force-publish=\* "$FEDERATION_RELEASE_VERSION"` ([more info](#lerna-version))
+    1. `git show --pretty="" --name-only HEAD` to see the files in the commit
+    1. `git --no-pager tag --points-at HEAD` to see the tags that were created
+1. `git push --follow-tags origin` to push the version bumps & tags created by lerna in the previous step
+1. `APOLLO_DIST_TAG=next npm run release:start-ci-publish` ([more info](#publishing))
+1. `open https://app.circleci.com/pipelines/github/apollographql/federation?filter=mine` to approve publishing to NPM
+    1. There will also be a message posted to #team-atlas in slack that has a link to the approval job
+1. `npm view @apollo/subgraph versions` to ensure the version that was just published shows up in the NPM registry
+
+#### Merge the release PR
+
+1. Merge the release PR to `main`
+
+![celebrate](https://media.giphy.com/media/LZElUsjl1Bu6c/giphy.gif)
+
+---
+
+## Changelogs
+
+Changelogs should be updated when creating a release. If there are no changes for a given package, add an entry that states `Released in sync with other federation packages but no changes to this package.`
+
+1. [gateway-js/CHANGELOG.md](gateway-js/CHANGELOG.md) all changes should be reflected in here, since it's essentially a rollup of all packages
```

Comment:
I'm conflicted on this one but I don't disagree. This is the most interesting changelog to users. There are probably changes to internals, etc. which don't need to roll up to this changelog though, so maybe per discretion?

Also `subgraph-js` is an exception to this.

Reply:
You're right, this was an overly broad generalization to `DEVELOPMENT.md`.

Though it seems like that doc says we should especially be rolling up `subgraph-js`

https://github.com/apollographql/federation/blob/0fb2212b83d46110afc5cff8f82be65458e5742d/DEVELOPMENT.md?plain=1#L35-L37

Reply:
This one's on me. `@apollo/subgraph` was once `@apollo/federation` which included both subgraph and composition related bits. It's the composition bits which used to be a part of the gateway (though that's also barely true anymore). I think this note can be removed and I'd standby `@apollo/subgraph` being independent of the others. With fed2 GA it might even make sense to separate this package from the monorepo...maybe.